pyam/iiasa.py

```diff
@@ -578,3 +579,60 @@ def read_iiasa(name, default=True, meta=True, creds=None, base_url=_AUTH_URL, **
         Arguments for :meth:`pyam.iiasa.Connection.query`
     """
     return Connection(name, creds, base_url).query(default=default, meta=meta, **kwargs)
+
+
+def lazy_read_iiasa(
+    file, name, default=True, meta=True, creds=None, base_url=_AUTH_URL, **kwargs
+):
+    """Check if the file in a given location is an up-to-date version of an IIASA
+    database. If so, load it. If not, load  data from IIASA scenario explorer and
+    save to that location. Does not check that the previously read version is a complete
+    instance of the database.
+
+    Parameters
+    ----------
+    file : str
+        The location to test for valid data and save the data if not up-to-date.
+    name : str
+        A valid name of an IIASA scenario explorer instance,
+        see :attr:`pyam.iiasa.Connection().valid_connections`
+    default : bool, optional
+        Return *only* the default version of each scenario.
+        Any (`model`, `scenario`) without a default version is omitted.
+        If :obj:`False`, return all versions.
+    meta : bool or list of strings, optional
+        If `True`, include all meta categories & quantitative indicators
+        (or subset if list is given).
+    creds : str, :class:`pathlib.Path`, list-like, or dict, optional
```

Comment:
Passing credentials directly is not supported any more.
```suggestion
    creds : str or :class:`pathlib.Path`, optional
```